integration/business-process-tests/tests-integration/bpel/src/test/java/org/wso2/ei/businessprocess/integration/tests/bpel/upload/BpelVersioningTest.java

```diff
@@ -110,8 +116,17 @@ public void checkVersion() throws InterruptedException, XMLStreamException, Remo
         uploadBpelForTest("HelloWorld2", artifactLocation);
         List<String> processAfter = null;
         for (int a = 0; a <= 10; a++) {
-            Thread.sleep(5000);
+
+            Awaitility.await()
+                    .pollInterval(50, TimeUnit.MILLISECONDS)
+                    .atMost(1000, TimeUnit.SECONDS)
+                    .until(isInforListAvailable("HelloWorld2"));
             processAfter = bpelProcessManagementClient.getProcessInfoList("HelloWorld2");
+
+            Awaitility.await()
+                    .pollInterval(20, TimeUnit.MILLISECONDS)
+                    .atMost(1000, TimeUnit.SECONDS)
```

Comment:
let's add constants for time periods since we have reused

Reply:
Fixed. Please refer latest commit.